homeassistant/components/sensor/coinmarketcap.py

```diff
@@ -0,0 +1,128 @@
+from homeassistant.helpers.entity import Entity
+from homeassistant.util import Throttle
+
+REQUIREMENTS = ['coinmarketcap==2.0.1']
+
+_LOGGER = logging.getLogger(__name__)
+
+ATTR_24H_VOLUME_USD = '24h_volume_usd'
+ATTR_AVAILABLE_SUPPLY = 'available_supply'
+ATTR_MARKET_CAP = 'market_cap_usd'
+ATTR_NAME = 'name'
+ATTR_PERCENT_CHANGE_24H = 'percent_change_24h'
+ATTR_PERCENT_CHANGE_7D = 'percent_change_7d'
+ATTR_PRICE = 'price_usd'
+ATTR_SYMBOL = 'symbol'
+ATTR_TOTAL_SUPPLY = 'total_supply'
+
+CONF_CURRENCY = 'currency'
+
+DEFAULT_CURRENCY = 'bitcoin'
+
+ICON = 'mdi:currency-usd'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(minutes=15)
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Optional(CONF_CURRENCY, default=DEFAULT_CURRENCY): cv.string,
+})
+
+
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the CoinMarketCap sensor."""
+    from coinmarketcap import Market
+    currency = config.get(CONF_CURRENCY)
+
+    try:
+        Market().ticker(currency)
+    except HTTPError:
+        _LOGGER.error('Currency "%s" is not available. Using "bitcoin"',
+                      currency)
+        currency = DEFAULT_CURRENCY
+
+    data = CoinMarketCapData(currency)
+
+    add_devices([CoinMarketCapSensor(data)])
+
+
+# pylint: disable=too-few-public-methods
+class CoinMarketCapSensor(Entity):
+    """Representation of a CoinMarketCap sensor."""
+
+    def __init__(self, data):
+        """Initialize the sensor."""
+        self.data = data
+        self._state = None
+        self._unit_of_measurement = 'USD'
+        self.update()
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return self.ticker.get('name')
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        return self.ticker.get('price_usd')
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit the value is expressed in."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the icon to use in the frontend, if any."""
+        return ICON
+
+    @property
+    def state_attributes(self):
+        """Return the state attributes of the sensor."""
+        return {
+            ATTR_24H_VOLUME_USD: self.ticker.get('24h_volume_usd'),
+            ATTR_AVAILABLE_SUPPLY: self.ticker.get('available_supply'),
+            ATTR_MARKET_CAP: self.ticker.get('market_cap_usd'),
+            ATTR_PERCENT_CHANGE_24H: self.ticker.get('percent_change_24h'),
+            ATTR_PERCENT_CHANGE_7D: self.ticker.get('percent_change_7d'),
+            ATTR_SYMBOL: self.ticker.get('symbol'),
+            ATTR_TOTAL_SUPPLY: self.ticker.get('total_supply'),
+        }
+
+    # pylint: disable=too-many-branches
+    def update(self):
+        """Get the latest data and updates the states."""
+        self.data.update()
+        self.ticker = json.loads(
+            self.data.ticker.decode('utf-8').strip('\n        '))[0]
```

Comment:
strip defaults to white space characters (including new lines and spaces). If you pass in characters, any of those in any order will be removed so no need to add 'space' multiple times.